src/store/hotspots/hotspotsSlice.ts

```diff
@@ -275,6 +283,53 @@ export const unfollowHotspot = createAsyncThunk<Hotspot[], string>(
   },
 )
 
+export enum SyncStatus {
+  full,
+  partial,
+  none,
+}
+
+type HotspotStatus = {
+  status: SyncStatus
+  percent: number
+}
+
+type StatusAttrs = {
+  address: string
+  statusTime?: string
+  blockHeight?: number
+}
+
+export type HotspotSyncCache = CacheRecord<HotspotStatus>
+export const fetchSyncStatus = createAsyncThunk(
+  'hotspotDetails/fetchHotspotData',
+  async ({ address, statusTime, blockHeight }: StatusAttrs, { getState }) => {
+    const {
+      hotspots: { syncStatuses },
+    } = getState() as {
+      hotspots: {
+        syncStatuses: Record<string, CacheRecord<HotspotStatus>>
+      }
+    }
+
+    if (syncStatuses[address] && hasValidCache(syncStatuses[address])) {
+      throw new Error(`sync status for hotspot ${address} is already valid`)
+    }
+
+    if (!statusTime) {
```

Comment:
status timestamp will be null until it has been gossiped. do we want this to throw an error in that case? or just consider it syncing?

Reply:
I'd consider it syncing

Reply:
So, like this?
```ts
    if (!statusTime) {
      return { status: SyncStatus.none, percent: 0 }
    }
```